internal/server/service_rotate_roots.go

```diff
@@ -18,7 +19,15 @@ func RotateRoots(ctx context.Context, workerAuthRepo *WorkerAuthRepositoryStorag
 		return errors.New(ctx, errors.InvalidParameter, op, "missing workerAuthRepo")
 	}
 
-	_, err := rotation.RotateRootCertificates(ctx, workerAuthRepo, opt...)
+	// Get current root certs so we can pass along the version
+	rootCerts := &types.RootCertificates{Id: CA_ID}
+	err := workerAuthRepo.Load(ctx, rootCerts)
+	if err != nil && err != nodeenrollment.ErrNotFound {
+		return errors.Wrap(ctx, err, op)
+	}
+	opt = append(opt, nodeenrollment.WithState(rootCerts.GetState()))
+
+	_, err = rotation.RotateRootCertificates(ctx, workerAuthRepo, opt...)
```

Comment:
`rotation.RotateRootCertificates` will return the new root certificates on success. It might be nice to return those from here, so that up in the service you don't need to rotate followed by a lookup.

Reply:
thanks!